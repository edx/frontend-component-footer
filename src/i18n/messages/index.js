import arMessages from './ar.json';
// no need to import en messages-- they are in the defaultMessage field
import es419Messages from './es_419.json';
import frMessages from './fr.json';
import zhcnMessages from './zh_CN.json';

const messages = {
  ar: arMessages,
  es: es419Messages,
  fr: frMessages,
  zh: zhcnMessages,
};

export default messages;
