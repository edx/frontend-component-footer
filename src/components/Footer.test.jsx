import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import Footer from './Footer';

describe('<Footer />', () => {
  describe('renders correctly', () => {
    it('renders without a language selector', () => {
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Footer />
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders without a language selector in es', () => {
      const tree = renderer
        .create((
          <IntlProvider locale="es">
            <Footer />
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders with a language selector', () => {
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Footer
              onLanguageSelected={() => {}}
              supportedLanguages={[
                { label: 'English', value: 'en' },
                { label: 'Español', value: 'es' },
              ]}
            />
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('handles language switching', () => {
    it('calls onLanguageSelected prop when a language is changed', () => {
      const mockHandleLanguageSelected = jest.fn();
      const wrapper = mount((
        <IntlProvider locale="en">
          <Footer
            onLanguageSelected={mockHandleLanguageSelected}
            supportedLanguages={[
              { label: 'English', value: 'en' },
              { label: 'Español', value: 'es' },
            ]}
          />
        </IntlProvider>
      ));

      wrapper.find('form').simulate('submit', {
        target: {
          elements: {
            'site-footer-language-select': {
              value: 'es',
            },
          },
        },
      });

      expect(mockHandleLanguageSelected).toHaveBeenCalledWith('es');
    });
  });
});
