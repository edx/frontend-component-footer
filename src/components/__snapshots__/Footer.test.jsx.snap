// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly renders with a language selector 1`] = `
<footer
  aria-label="edX Home"
  className="footer d-flex border-top py-3 px-4"
  role="contentinfo"
>
  <div
    className="container-fluid d-flex"
  >
    <a
      aria-label="edX Home"
      className="d-block"
      href="https://open.edx.org"
    >
      <img
        alt="Powered by Open edX"
        src="https://edx-cdn.org/v3/default/logo-trademark.svg"
        style={
          Object {
            "maxHeight": 45,
          }
        }
      />
    </a>
    <div
      className="flex-grow-1"
    />
    <form
      className="form-inline"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="d-inline-block m-0"
          htmlFor="site-footer-language-select"
        >
          <span>
            Choose Language
          </span>
        </label>
        <select
          className="form-control-sm mx-2"
          defaultValue="en"
          id="site-footer-language-select"
          name="site-footer-language-select"
        >
          <option
            value="en"
          >
            English
          </option>
          <option
            value="es"
          >
            Espa√±ol
          </option>
        </select>
        <button
          className="btn btn-outline-primary btn-sm"
          type="submit"
        >
          <span>
            Apply
          </span>
        </button>
      </div>
    </form>
  </div>
</footer>
`;

exports[`<Footer /> renders correctly renders without a language selector 1`] = `
<footer
  aria-label="edX Home"
  className="footer d-flex border-top py-3 px-4"
  role="contentinfo"
>
  <div
    className="container-fluid d-flex"
  >
    <a
      aria-label="edX Home"
      className="d-block"
      href="https://open.edx.org"
    >
      <img
        alt="Powered by Open edX"
        src="https://edx-cdn.org/v3/default/logo-trademark.svg"
        style={
          Object {
            "maxHeight": 45,
          }
        }
      />
    </a>
    <div
      className="flex-grow-1"
    />
  </div>
</footer>
`;

exports[`<Footer /> renders correctly renders without a language selector in es 1`] = `
<footer
  aria-label="edX Home"
  className="footer d-flex border-top py-3 px-4"
  role="contentinfo"
>
  <div
    className="container-fluid d-flex"
  >
    <a
      aria-label="edX Home"
      className="d-block"
      href="https://open.edx.org"
    >
      <img
        alt="Powered by Open edX"
        src="https://edx-cdn.org/v3/default/logo-trademark.svg"
        style={
          Object {
            "maxHeight": 45,
          }
        }
      />
    </a>
    <div
      className="flex-grow-1"
    />
  </div>
</footer>
`;
